pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -224,6 +225,21 @@ Map<String, DefaultColumnAction> computeDefaultColumnActionMap() {
       if (columnMetadata != null) {
         // Column exists in the segment, check if we need to update the value.
 
+        if (_segmentWriter != null && !columnMetadata.isAutoGenerated()) {
+          // Check that forward index disabled isn't enabled / disabled on an existing column (not auto-generated).
+          // TODO: Add support for reloading segments when forward index disabled flag is enabled or disabled
+          boolean forwardIndexDisabled = !_segmentWriter.hasIndexFor(column, ColumnIndexType.FORWARD_INDEX);
+          if (forwardIndexDisabled != _indexLoadingConfig.getForwardIndexDisabledColumns()
+              .contains(column)) {
+            String failureMessage =
+                "Forward index disabled in segment: " + forwardIndexDisabled + " for column: " + column
+                    + " does not match forward index disabled flag: "
+                    + _indexLoadingConfig.getForwardIndexDisabledColumns().contains(column) + " in the TableConfig, "
+                    + "updating this flag is not supported at the moment.";
```

Comment:
Not just updating / toggling, even `setting` this flag for the very first time for an existing column is not supported (yet) on the reload path. 

I think we should also log a warning for now and ignore and until the reload support is added, anyone who wants to leverage this feature on an existing column should consider backfilling / refresh

Reply:
I've modified the comment but I still think we should fail and throw an exception here. let's discuss more about this if you think a warning log is enough.